src/System.Management.Automation/utils/Telemetry.cs

```diff
@@ -0,0 +1,403 @@
+                case "yes":
+                    return true;
+                case "false":
+                case "0":
+                case "no":
+                    return false;
+                default:
+                    return defaultValue;
+            }
+        }
+
+        /// <summary>
+        /// Send telemetry as a metric.
+        /// </summary>
+        /// <param name="metricId">The type of telemetry that we'll be sending.</param>
+        /// <param name="data">The specific details about the telemetry.</param>
+        internal static void SendTelemetryMetric(TelemetryType metricId, string data)
+        {
+            if (!CanSendTelemetry)
+            {
+                return;
+            }
+
+            string metricName = metricId.ToString();
+            string uuidString = s_uniqueUserIdentifier;
+            try
+            {
+                switch (metricId)
+                {
+                    case TelemetryType.ApplicationType:
+                    case TelemetryType.PowerShellCreate:
+                    case TelemetryType.RemoteSessionOpen:
+                    case TelemetryType.ExperimentalEngineFeatureActivation:
+                        s_telemetryClient.GetMetric(metricName, "uuid", "SessionId", "Detail").TrackValue(metricValue: 1.0, uuidString, s_sessionId, data);
+                        break;
+                    case TelemetryType.ExperimentalModuleFeatureActivation:
+                        string experimentalFeatureName = GetExperimentalFeatureName(data);
+                        s_telemetryClient.GetMetric(metricName, "uuid", "SessionId", "Detail").TrackValue(metricValue: 1.0, uuidString, s_sessionId, experimentalFeatureName);
+                        break;
+                    case TelemetryType.ModuleLoad:
+                        string moduleName = GetModuleName(data); // This will return anonymous if the modulename is not on the report list
+                        s_telemetryClient.GetMetric(metricName, "uuid", "SessionId", "Detail").TrackValue(metricValue: 1.0, uuidString, s_sessionId, moduleName);
+                        break;
+                }
+            }
+            catch
+            {
+                // do nothing, telemetry can't be sent
+                // don't send the panic telemetry as if we have failed above, it will likely fail here.
+            }
+
+        }
+
+        // Get the experimental feature name. If we can report it, we'll return the name of the feature, otherwise, we'll return "anonymous"
+        private static string GetExperimentalFeatureName(string featureNameToValidate)
+        {
+            // An experimental feature in a module is guaranteed to start with the module name
+            // we can strip out the text past the last '.' as the text before that will be the ModuleName
+            int lastDotIndex = featureNameToValidate.LastIndexOf('.');
+            string moduleName = featureNameToValidate.Substring(0,lastDotIndex);
```

Comment:
> 0,lastDotIndex

Can you please add a space after the comma? CodeFacotr might complain :)

Reply:
fixed